### PR TITLE
[BUGFIX] Return 401 when user does not exist, add test to back it up

### PR DESCRIPTION
**Pull request by nigelmann**
This solves #69 (nice)

### CODE REVIEW

File: apps/backend/src/main/java/ch/heig/pdg/backend/exception/exceptions/UnauthorizedException.java [hunk 1]
```diff
@@ -1,6 +1,9 @@
 package ch.heig.pdg.backend.exception.exceptions;
 
 public class UnauthorizedException extends RuntimeException {
+    public UnauthorizedException() {
```
Comment (noahboegli) [inline apps/backend/src/main/java/ch/heig/pdg/backend/exception/exceptions/UnauthorizedException.java:4]: I understand the idea of being willing to use `orElseThrow`. But wouldn't it make sense to at least notify why the 401 happened ? I.e. being able to separate between unexisting username and wrong password ?

File: apps/backend/src/main/java/ch/heig/pdg/backend/presentation/AuthenticationController.java [hunk 2]
```diff
@@ -30,7 +30,7 @@ public AuthenticationController(UserRepository userRepository, JWTGeneratorServi
 
     @Override
     public ResponseEntity<JWTDTO> getToken(CredentialsDTO credentialsDTO) {
-        User user = this.userRepository.findByUsername(credentialsDTO.getUsername()).orElseThrow(NotFoundException::new);
+        User user = this.userRepository.findByUsername(credentialsDTO.getUsername()).orElseThrow(UnauthorizedException::new);
```
Comment (noahboegli) [inline apps/backend/src/main/java/ch/heig/pdg/backend/presentation/AuthenticationController.java:33]: As per said above `orElseThrow(() -> new UnauthorizedException("wrong user")`

File: apps/backend/src/main/java/ch/heig/pdg/backend/exception/exceptions/UnauthorizedException.java (hunk 1 above)
Comment (nigelmann) [reply apps/backend/src/main/java/ch/heig/pdg/backend/exception/exceptions/UnauthorizedException.java:4]: I do not agree a failed login must be obfuscated if it's the password or the username the problem.

File: apps/backend/src/main/java/ch/heig/pdg/backend/presentation/AuthenticationController.java (hunk 2 above)
Comment (nigelmann) [reply apps/backend/src/main/java/ch/heig/pdg/backend/presentation/AuthenticationController.java:33]: No

Comment (HugoMendes98) [reply apps/backend/src/main/java/ch/heig/pdg/backend/presentation/AuthenticationController.java:33]: Both are correct.

Must of the time, the _frontend_ does not care to know why the login failed (our case).

But depending on the expected "login-flow", the reason should be sent.
EG: _Google_ or _Microsoft_ = first the email -> the password

For this project, KIS (Keep It Simple).
(Return me if the password is already used)

**Review by HugoMendes98 — APPROVED** [review]

**Comment by noahboegli** [comment]
Let's remove passwords altogether and instead just use the username

**Review by noahboegli — APPROVED** [review]
As per discussed off grid, all good for me

**Pull request merged**